docs/resource/README.md

```diff
@@ -99,10 +99,35 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
 <!-- semconv service_experimental -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
-| `service.instance.id` | string | The string ID of the service instance. [1] | `my-k8s-pod-deployment-1`; `627cc493-f310-47de-96bd-71410b7dec09` | Recommended |
+| `service.instance.id` | string | The string ID of the service instance. [1] | `627cc493-f310-47de-96bd-71410b7dec09` | Recommended |
 | `service.namespace` | string | A namespace for `service.name`. [2] | `Shop` | Recommended |
 
-**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations).
+**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words
+`service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to
+distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled
+service).
+
+Implementations are recommended to generate a random Version 1 or Version 4 [RFC
```

Comment:
What is an implementation in this context? On first read I read it as an OpenTelemetry SDK, but the following phrase made me think it was addressing end users instead (or maybe as well?):

> but are free to use an inherent unique ID as the source of this value if stability is desirable

I think we need to make two things clear:
- Users can and should set `service.instance.id` if they desire a stable value
- SDK implementations SHOULD provide a random fallback v1/v4 uuid if the user hasn't already provided a value

WDYT? Do you agree with this? Do you think the text already implies that and I'm reading into it too much?

Reply:
> SDK implementations SHOULD provide a random fallback v1/v4 uuid if the user hasn't already provided a value

The reason I want to make this clear is that in the past I've tried to add fallback random uuid `service.instance.id` to the opentelemetry java SDK but received pushed back that the spec / semantic conventions is ambiguous abut this. I want to make sure that we're very clear that this is the intent.

Reply:
An implementation here is typically an SDK, but might be a Collector (if it has enough information to make a good service.instance.id). The "inherent unique ID" phrasing is because some platforms might have an ID that is unique to that execution (like lambdas) or unique to the workload (like the Erlang example that @jsuereth mentioned elsewhere).

In the typical case, SDKs will be generating UUIDs.